ios/Surveys/Sources/Presentation/Modules/Home/HomeView.swift

```diff
@@ -10,12 +10,38 @@ import SwiftUI
 
 struct HomeView: View {
 
+    @StateObject var viewModel = HomeViewModel()
+
     var body: some View {
+        switch viewModel.state {
+        case .idle:
+            homeContent()
+                .onAppear {
+                    viewModel.loadProfile()
+                }
+        case .loading:
+            homeContent(isLoading: true)
+        case .loaded:
+            homeContent()
+        case let .failure(message):
+            homeContent()
+                .alert(isPresented: .constant(true), content: {
+                    Alert(
+                        title: Text(Localize.generalTextSurveys()),
+                        message: Text(message),
+                        dismissButton: .default(Text(Localize.generalButtonGotIt()))
+                    )
+                })
+        }
+    }
+
+    private func homeContent(isLoading: Bool = false) -> some View {
         VStack {
             // TODO: - Demo data for POW
```

Comment:
I will remove the TODO as I've finished the integrate task.

For the background color at line 49, I will remove it when we have the list of the survey because all the text colors of the header are white.

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/99/commits/baa0317a7b935a87006a9cf09e36fd00e6f988e3